content/tutorials/MQTT-snake-js.textile

```diff
@@ -0,0 +1,267 @@
+
+Now that you have a method to publish through MQTT, you need to use it to send our key presses. Replace the contents of the function associated with @process.stdin.on('keypress')@ with the following:
+
+```[javascript]
+  if (key) {
+    if (key.ctrl && (key.name == 'c')) {
+      process.stdin.pause();
+      client.end();
+    } else if(key.name == 'left') {
+      publishMessage('input', 'left');
+    } else if(key.name == 'right') {
+      publishMessage('input', 'right');
+    } else if(key.name == 'up') {
+      publishMessage('input', 'up');
+    } else if(key.name == 'down') {
+      publishMessage('input', 'down');
+    } else if(key.name == 'return') {
+      publishMessage('input', 'return');
+    }
+  }
+```
+
+This will use the @publishMessage@ function you created to send the key presses you're interested in. If a user types ctrl-c, the stream will be paused and the client's connection will be closed so that the program can stop. Otherwise, you're sending the key pressed to the channel @input@.
+
+"See this step in Github":https://github.com/ably/tutorials/commit/9f8db5437e4730d8776a63f95dc7820ae778d7f1
+
+h2. Step 7 - Receiving inputs from Ably
+
+Now that you've set up your controller, it's time to create something to run the game of Snake, and receive the data from Ably that you sent from the controller. You'll be using the normal Ably Realtime protocol for this part. This is possible due to the fact Ably is "Protocol Agnostic":https://support.ably.io/solution/articles/3000054126-do-you-support-mqtt-are-you-protocol-agnostic, allowing protocols to be used interchangeably.
+
+Firstly create a file called @snake.html@, and place the following code inside:
+
+```[html]
+  <!doctype html>
+  <html>
+    <head>
+      <title>Ably Realtime Snake</title>
+      <meta name="description" content="Fun Snake Game by Ably">
+      <meta name="keywords" content="ably realtime snake">
+      <link rel="stylesheet" href="snake.css">
+      <script src="http://cdn.ably.io/lib/ably.min-1.js"></script>
+    </head>
+    <body>
+      <script src="snake.js"></script>
+      <h1 id="heading">Ably Realtime Snake</h1>
+      <canvas id="gc" height="400" width="400"></canvas>
+      </table>
+    </body>
+  </html>
+```
+
+Next, create another file called @snake.css@ containing the following:
+
+```[css]
+#heading {
+	color: #FFA500;
+	text-align: center;
+}
+
+#gc {
+    padding-left: 0;
+    padding-right: 0;
+    margin-left: auto;
+    margin-right: auto;
+    display: block;
+    width: "400";
+    height: "400";
+}
+```
+
+Now, create a file called @snake.js@, and put in the following code, replacing REPLACE_WITH_YOUR_API_KEY with your API Key:
+
+```[javascript]
+  var ably = new Ably.Realtime('REPLACE_WITH_YOUR_API_KEY');
+  var enc = new TextDecoder();
+  var channel = ably.channels.get('input');
+  channel.subscribe(function(message) {
+    var command = enc.decode(message.data);
+    console.log(command);
+  });
+```
```

Comment:
might be worth a line or two of explanation about why you need the TextDecoder, when you wouldn't with a native Ably client lib (mqtt being a binary protocol with no encoding, the payload is just raw bits that it's up to the receiver to interpret)